### PR TITLE
VEOSS-1213 initial commit

### PR DESCRIPTION
**Pull request by kapoor-prateek**

### CODE REVIEW

File: app/views/IndexView.scala.html [hunk 1]
```diff
@@ -20,120 +20,142 @@
...
+    <span class="govuk-caption-m">@messages("index.vrn", vrn)</span>
+</h1>
 
-    <p class="govuk-body">@messages("index.paragraph")</p>
+<div class="govuk-warning-text">
+    <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
+    <strong class="govuk-warning-text__text">
+        <span class="govuk-warning-text__assistive">Warning</span>
+        @if(excludedTrader.nonEmpty) {
+            @if(excludedTrader.get.exclusionReason == 4) {
```
Comment (DevAndyroo) [inline app/views/IndexView.scala.html:72]: You can simplify this by using a case/match statement

File: conf/messages.en [hunk 2]
```diff
@@ -508,3 +508,7 @@ whichVatPeriodToPay.error.required = Select a period to pay
 noPayments.title = You do not owe any One Stop Shop VAT
 noPayments.heading = You do not owe any One Stop Shop VAT
 noPayments.textWithLink = Return to <a href={0}>your account</a>.
+
+warning.exclusion.message1 = You have left this service, but you must complete and pay your final return.
+warning.exclusion.message2 = We''ve removed you from this service, but you must complete and pay your final return.
+warning.exclusion.message4 = We''ve removed you from this service, but you must complete and pay your final return. You cannot re-join until [{0}]
```
Comment (DevAndyroo) [inline conf/messages.en:514]: can we use more descriptive labels than 1, 2, 4?

File: app/views/IndexView.scala.html [hunk 3]
```diff
@@ -20,120 +20,142 @@
...
+excludedTrader: Option[ExcludedTrader]
 )(implicit request: Request[_], messages: Messages)
 
 @layout(
-    pageTitle = titleNoForm(messages("index.title")),
-    showBackLink = false,
-    fullWidth = true
+pageTitle = titleNoForm(messages("index.title")),
+showBackLink = false,
+fullWidth = true
```
Comment (DevAndyroo) [inline app/views/IndexView.scala.html:42]: Most of this file has gone skewiff with the indentation

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/views/IndexView.scala.html (hunk 1 above)
Comment (kapoor-prateek) [reply app/views/IndexView.scala.html]: If statement is simplified with case/match statement

File: conf/messages.en (hunk 2 above)
Comment (kapoor-prateek) [reply conf/messages.en]: Renamed labels

File: app/views/IndexView.scala.html (hunk 3 above)
Comment (kapoor-prateek) [reply app/views/IndexView.scala.html]: code formatted

File: app/controllers/YourAccountController.scala [hunk 4]
```diff
@@ -26,15 +26,16 @@ import models.{Period, UserAnswers}
...
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import viewmodels.yourAccount._
 import views.html.IndexView
-
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class YourAccountController @Inject()(
                                        cc: AuthenticatedControllerComponents,
+                                       service: ExclusionService,
```
[the remark this answers is not in the document]
Comment (kapoor-prateek) [reply app/controllers/YourAccountController.scala:38]: Renamed service -> exclusionService

File: conf/messages.en [hunk 5]
```diff
@@ -508,3 +508,11 @@ whichVatPeriodToPay.error.required = Select a period to pay
...
 noPayments.heading = You do not owe any One Stop Shop VAT
 noPayments.textWithLink = Return to <a href={0}>your account</a>.
+
+warning.exclusion.trader = You have left this service, but you must complete and pay your final return.
+warning.exclusion.hmrc = We''ve removed you from this service, but you must complete and pay your final return.
+warning.exclusion.quarantineUser = We''ve removed you from this service, but you must complete and pay your final return. You cannot re-join until [{0}].
+warning.leftService = You have left this service.
+previous.return.correct = You can correct previous returns directly with the country where you made the sales.
+return.completed = You''ve completed your final return.
+oss.account.message = If you have any questions about your OSS account, contact <a class="govuk-link" href="mailto:vatoss.contact@hmrc.gov.uk">vatoss.contact@hmrc.gov.uk</a>
```
[the remark this answers is not in the document]
Comment (kapoor-prateek) [reply conf/messages.en]: Appended with yourAccount

File: conf/messages.en [hunk 6]
```diff
@@ -508,3 +508,15 @@ whichVatPeriodToPay.error.required = Select a period to pay
...
+excludedNotPermitted.title = You cannot use this service
+excludedNotPermitted.heading = You cannot use this service
+excludedNotPermitted.link = Back to your account.
+
+yourAccount.warning.exclusion.trader = You have left this service, but you must complete and pay your final return.
+yourAccount.warning.exclusion.hmrc = We''ve removed you from this service, but you must complete and pay your final return.
+yourAccount.warning.exclusion.quarantineUser = We''ve removed you from this service, but you must complete and pay your final return. You cannot re-join until [{0}].
+yourAccount.warning.leftService = You have left this service.
+yourAccount.previous.return.correct = You can correct previous returns directly with the country where you made the sales.
+yourAccount.return.completed = You''ve completed your final return.
```
Comment (DevAndyroo) [inline conf/messages.en:521]: can we name this `yourAccount.finalReturn.completed`

File: app/views/IndexView.scala.html [hunk 7]
```diff
@@ -39,33 +42,81 @@
...
+<h1 class="govuk-heading-l">
+    @messages("index.heading")
+    <span class="govuk-caption-m">@businessName</span>
+    <span class="govuk-caption-m">@messages("index.vrn", vrn)</span>
+</h1>
+
+@if(exclusionsEnabled) {
+    <div class="govuk-warning-text">
+        <strong class="govuk-warning-text__text">
+            <span class="govuk-warning-text__assistive">Warning</span>
```
Comment (DevAndyroo) [inline app/views/IndexView.scala.html:70]: I'd like a demo of this page, with the following scenario:

- exclusions enabled
- user is not excluded

I feel like this part will still be shown

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

**Review by DevAndyroo — APPROVED** [review]

**Pull request merged**